Client/Frontend/Fakespot/FakespotViewController.swift

```diff
@@ -128,6 +127,13 @@ class FakespotViewController: UIViewController, Themeable, Notifiable, UIAdaptiv
         viewModel.isSwiping = false
     }
 
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        guard #available(iOS 15.0, *) else { return }
+        let currentDetent = viewModel.getCurrentDetent(for: presentationController)
+        viewModel.recordBottomSheetDisplayed(in: currentDetent == .large ? .fullView : .halfView)
```

Comment:
I think it would be better to have all of this logic in the viewModel and pass the presentationController into the method that handles the telemetry recording.